File: python/exercise34/employee_list_removal.py
with open("./employees.txt") as file:
    employees = [file.strip("\n") for file in file.readlines()]


print(f"There are {len(employees)} employees:")
for emp in employees:
    print(f"  - {emp.strip()}")


name = input("\nEnter an employee name to remove: ")
if name not in employees:
    print(f"{name} is not an employee...")
    exit(0)

employees.remove(name)

print(f"\nThere are {len(employees)} employees:")
for emp in employees:
    print(f"  - {emp}")

with open("./employees.txt", "w") as file:
    file.writelines(employees)

File: javascript/exercise34/employee-list-removal.js
import fs from "node:fs";
import { input } from "pynput";

let employees = read_employees().split("\n");
display_employees(employees);

const name = await input("Enter an employee name to remove: ");
const nameIdx = employees.indexOf(name);

if (nameIdx < 0) {
	console.error(`"${name}" is not found`);
	process.exit(0);
}

employees = Array().concat(
	employees.slice(0, nameIdx),
	employees.slice(nameIdx + 1),
);

display_employees(employees);

function read_employees() {
	const filename = "./employees.txt";

	const buffer = fs.readFileSync(filename);
	return buffer.toString().trim();
}

/** @param {string[]} names */
function display_employees(names) {
	console.log(`There are ${names.length} employees:`);
	for (const name of names) {
		console.log(name);
	}
	console.log();
}
